x-pack/plugins/stack_connectors/server/connector_types/lib/servicenow/service.ts

```diff
@@ -249,6 +255,72 @@ export const createExternalService: ServiceFactory = ({
     }
   };
 
+  const getIncidentByCorrelationId = async (correlationId: string): Promise<ServiceNowIncident> => {
+    try {
+      const res = await request({
+        axios: axiosInstance,
+        url: getIncidentByCorrelationIdUrl(correlationId),
+        method: 'get',
+        logger,
+        configurationUtilities,
+      });
+
+      checkInstance(res);
+
+      if (res.status === 404) {
+        logger.warn(`No incident found with correlation ID ${correlationId}.`);
+      }
+
+      const foundIncident = res.data.result[0] ?? null;
+
+      return foundIncident;
+    } catch (error) {
+      throw createServiceError(error, `Unable to get incident by correlation ID ${correlationId}`);
+    }
+  };
+
+  const closeIncident = async (params: ExternalServiceParamsClose) => {
+    try {
+      const { correlationId, incidentId } = params;
+      let incidentToBeClosed = null;
+
+      if (correlationId == null && incidentId == null) {
+        throw new Error('No correlationId or incidentId found.');
+      } else if (incidentId) {
+        incidentToBeClosed = await getIncident(incidentId);
+      } else if (correlationId) {
+        incidentToBeClosed = await getIncidentByCorrelationId(correlationId);
+      }
+
+      if (incidentToBeClosed?.state === 7) {
+        logger.warn(`Incident is already closed.`);
```

Comment:
I think we should enrich the log message to include more information. If you see the log without having any context it is difficult to understand what this message is about. What about `[ServiceNow][CloseIncident] Incident with correlation_id: ${correlationID} or incidentId: ${incidentId} is closed.`